### PR TITLE
RGW: Set appropriate bucket quota value (when quota value is less than 0)

### PR DESCRIPTION
**Pull request by GaryHyg**
When the bucket quota value is less than 0, change the quota value to -1

Signed-off-by: Garyhyg huygbj@inspur.com

### CODE REVIEW

**Comment by GaryHyg** [comment]
@tchaikov hi, tchaikov , could you take the time to check these changes, thank you!

**Comment by cbodley** [comment]
passed qa in http://pulpito.ceph.com/cbodley-2020-01-23_17:19:37-rgw-wip-cbodley-testing-distro-basic-smithi/

File: src/rgw/rgw_rest_user.cc [hunk 1]
```diff
@@ -183,6 +183,9 @@ void RGWOp_User_Create::execute()
   }
 
   if (max_buckets != default_max_buckets)
```
Comment (yehudasa) [inline src/rgw/rgw_rest_user.cc]: @cbodley note that if statement doesn't open a scope, so this is completely broken now

Comment (mattbenjamin) [reply src/rgw/rgw_rest_user.cc]: it makes the following statement depend on the condition;  the following statement is I think "if (max_buckets < 0) { max_buckets =-1; }"  then op_state.set_max_buckets(max_buckets); is executed unconditionally
no?\3

Comment (yehudasa) [reply src/rgw/rgw_rest_user.cc]: @mattbenjamin yup, this needs to be fixed.

Comment (tchaikov) [reply src/rgw/rgw_rest_user.cc]: just came across this. because the GCC warning. see #32884

**Pull request merged**